src/language_concepts/data_types.md

```diff
@@ -66,42 +68,54 @@ fn main(x : Field, y : u32) {
 }
 ```
 
-`x`, `y` and `z` are all private values in this example. However, `x` is a `Field` while `y` and `z` are unsigned 32-bit integers. If `y` or `z` exceeds the range \\([0,2^{32}-1]\\), proofs created will be rejected by the verifier.
+`x`, `y` and `z` are all private values in this example. However, `x` is a field while `y` and `z` are unsigned 32-bit integers. If `y` or `z` exceeds the range \\([0,2^{32}-1]\\), proofs created will be rejected by the verifier.
 
 > **Note:** The default TurboPlonk backend supports both even (e.g. `u16`, `u48`) and odd (e.g. `u5`, `u3`) sized integer types.
 
 ### The Boolean Type
 
-A boolean type in Noir has two possible values: `true` and `false`, and is specified using `bool`. For example:
+The `bool` type in Noir has two possible values: `true` and `false`:
 
 ```rust,noplaypen
 fn main() {
-    let t = true; // type implicitly interpreted
-    let f: bool = false; // type explicit annotated
+    let t = true;
+    let f: bool = false;
 }
 ```
 
 > **Note:** When returning a boolean value, it will show up as a value of 1 for `true` and 0 for `false` in _Verifier.toml_.
 
-The main usage of the boolean type is in conditionals (e.g. `if` expressions, `constrain`). More about conditionals are covered in the [Control Flow](./control_flow.md) and [Constrain Statement](./constrain.md) sections.
+The boolean type is most commonly used in conditionals like `if` expressions and `constrain` statements. More about conditionals is covered in the [Control Flow](./control_flow.md) and [Constrain Statement](./constrain.md) sections.
 
 ## Compound Types
 
 A compound type groups together multiple values into one type. Elements within a compound type can be private or public.
 
 ### The Array Type
 
-An array is one way of grouping together values into one compound type:
+An array is one way of grouping together values into one compound type. It can be implicitly inferred or explicitly defined with `[<Type>; <Size>]`:
```

Comment:
"Defined" is for defining a name to be equal to a value. Since there is no value here, it'd be more accurate to say "Array types can be be inferred or explicilty specified via the syntax `[<Type>; <Size>]`"